src/searcher.rs

```diff
@@ -164,6 +164,12 @@ impl Searcher {
         self.inner.num_docs()
     }
 
+    /// Returns the number of segments in the index.
+    #[getter]
+    fn num_segments(&self) -> usize {
```

Comment:
I did have to add this to expose the number of segments to the Python bindings to access it from my test.

It's not completely related but we do expose it already in `__repr__()` so I figured it may be useful anyways. If so desired, I can break this out in another change.